backend/src/api.py

```diff
@@ -114,10 +100,27 @@ def register(
         side_effects: bool = False,
         deprecated: bool = False,
         default_nodes: List[DefaultNode] | None = None,
+        decorators: List[Callable] | None = None,
     ):
         def inner_wrapper(wrapped_func: T) -> T:
             p_inputs, group_layout = _process_inputs(inputs)
-            p_output = _process_outputs(outputs)
+            p_outputs = _process_outputs(outputs)
+
+            TYPE_CHECK_LEVEL = os.environ.get("TYPE_CHECK_LEVEL", "none")
```

Comment:
Please do some more validation for `TYPE_CHECK_LEVEL`. Right now, literally any string that isn't `"none"` or `"warn"` will behave like `"error"`, e.g. `"WARN"`. Maybe we could add a function `def get_type_checking_level() -> Literal["none", "warn", "error"]`?